src/caffe/util/detectnet_coverage_rectangular.cpp

```diff
@@ -0,0 +1,466 @@
+      );
+      Point2v br(
+          transformedLabel.bottomRight_x + gridbox.tl().x,
+          transformedLabel.bottomRight_y + gridbox.tl().y
+      );
+
+      Scalar color;
+      if (!this->param_.visualize_obj_norm()) {
+        //define color of bounding box as a function of its bounding box and
+        color = this->bboxToColor(tl, br);
+        // coverage intensity:
+        color *= max_coverage;
+      }
+      else {
+        //if obj_norm, the redder the smaller cvg region
+        color = Scalar(0, 0, 255*5);
+        color *= transformedLabel.obj_norm;
+      }
+
+      Point2v center(
+          (gridbox.tl() + gridbox.br()) * 0.5
+      );
+
+
+      // draw coverage dot and bounding box:
+      circle(img_vis, center, 2, color, -1); //-1 means filled
+      rectangle(img_vis, tl, br, Scalar(0, 255, 0), 1);
+      // draw class label:
+      std::stringstream ss;
+      ss << max_coverage_index;
+      setLabel(img_vis, ss.str(), tl, 0.5);
+    } // foreach grid_x
+  } // foreach grid_y
+}
+
+
+template <typename Dtype>
+void CoverageGenerator<Dtype>::generate(
+    Dtype* transformedLabels,
+    const vector <BboxLabel>& _bboxList
+) const {
+  // ignore all label types we don't care about:
+  const vector<BboxLabel> bboxList(this->pruneBboxes(_bboxList));
+
+  //clear out transformed_label, things may remain inside from the last batch
+  this->clearLabel(transformedLabels);
+
+  //foreach bbox in list:
+  foreach_(const BboxLabel& label, bboxList) {
+    Rectv bbox(label.bbox);
+
+    //Define the area in which we intend to mark as containing an object to
+    // the network:
+    Rectv coverage(this->coverageBoundingBox(bbox));
+
+    // coverage region is implementation specific and defined by extending
+    //  classes, but must fit within coverage Rectf:
+    scoped_ptr<CoverageRegion> coverageRegion(
+        this->coverageRegion(coverage)
+    );
+    Dtype dObjNormValue = objectNormValue(*coverageRegion);
+
+    // This Rect includes all gridspaces which overlap the coverage rectangle:
+    Rect g_coverage(this->imageRectToGridRect(coverage));
+
+    for (size_t g_y = g_coverage.tl().y; g_y < g_coverage.br().y; g_y++) {
+      for (size_t g_x = g_coverage.tl().x; g_x < g_coverage.br().x; g_x++) {
+        // the current gridbox:
+        Rectv gridBox(
+          Point2v(g_x, g_y) * (float) this->param_.stride(),
+          Size2v(this->param_.stride(), this->param_.stride())
+        );
+
+        // the amount of the gridbox covered by the coverage region:
+        Dtype cvgArea =
+            coverageRegion->intersectionArea(gridBox)
+          / this->gridBoxArea_;
+
+        TransformedLabel tLabel(this->transformedLabelAt(
+            transformedLabels,
+            g_x,
+            g_y
+        ));
+
+        // maximum coverage value:
+        /*Dtype max_coverage = *std::max_element(
+            tLabel.coverages.begin(),
+            tLabel.coverages.end()
+        );*/
```

Comment:
I sort of gave up on cleaning up the code because there was such an enormous flood of lint errors. I resorted to just renaming files and layers and leaving the code internals as-is.

Obviously, removing a few comments is not a big deal, though. Does this need to be removed for IP reasons?